fe/fe-core/src/main/java/com/starrocks/qe/ConnectContext.java

```diff
@@ -520,14 +520,14 @@ public void kill(boolean killConnection) {
 
         if (killConnection) {
             isKilled = true;
+            // Now, cancel running process.
+            StmtExecutor executorRef = executor;
+            if (executorRef != null) {
+                executorRef.cancel();
```

Comment:
Why not guarantee the cancel() finished before close()?
Wait in the close() sounds not a good idea.